app/services/students/after_course_completion_service.rb

```diff
@@ -3,13 +3,29 @@ module Students
   class AfterCourseCompletionService
     def initialize(
       student,
-      notification_service: Developers::NotificationService.new
+      notification_service: Developers::NotificationService.new,
+      analytics: Segment::Analytics.new(
```

Comment:
And in services or other classes then we can just use the global object `SegmentAnalytics` for example that we set up in Initializer.

Reply:
Yeah, since we were not going to be working on this anymore I figured it wasn't worth it spending too much time on the implementation.

For the other projects that are more long term I used an adapter approach where the adapter is loaded during initialization ([like this](https://github.com/growthtribeacademy/gt-assessment-app/blob/432c6d285706c7f9504c29e4734139b7a97ac0ef/config/environments/development.rb#L67). The Segment analytics class is then injected as a dependency and we can then add other analytics tools to the adapter if needed and keep using the underlying implementation. The adapter can be used from anywhere using `Rails.configuration.analytics_adapter`.